src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ProductController.php

```diff
@@ -137,7 +137,7 @@ public function indexAction(Request $request, ProductFilters $filters): Response
             'categoryFilterForm' => $categoriesForm->createView(),
             'productGrid' => $this->presentGrid($productGrid),
             'enableSidebar' => true,
-            'layoutHeaderToolbarBtn' => $this->getProductToolbarButtons(),
+            'layoutHeaderToolbarBtn' => $this->getProductToolbarButtons($request->get('_legacy_controller')),
```

Comment:
Maybe add the permission needed as a parameter as well

Reply:
every button in toolbar may need different permission (though now there is only one button, but anyway, it wouldn't be right to pass one permission)

Reply:
OK